pkg/ipam/node.go

```diff
@@ -586,6 +586,30 @@ func (n *Node) removeStaleReleaseIPs() {
 	}
 }
 
+// abortNoLongerExcessIPs allows for aborting release of IP if new allocations on the node result in a change of excess
+// count or the interface selected for release.
+func (n *Node) abortNoLongerExcessIPs(excessMap *map[string]bool) {
+	n.mutex.Lock()
+	defer n.mutex.Unlock()
+	if n.resource.Status.IPAM.ReleaseIPs == nil || len(n.resource.Status.IPAM.ReleaseIPs) == 0 {
```

Comment:
No need for a nil check here, `len(nil)` is safe and will always return 0.
```suggestion
	if len(n.resource.Status.IPAM.ReleaseIPs) == 0 {
```